secretspec-derive/src/lib.rs

```diff
@@ -524,13 +524,14 @@ fn generate_secret_assignment(
 ) -> proc_macro2::TokenStream {
     if is_optional {
         quote! {
-            #field_name: #source.get(#secret_name).cloned()
+            #field_name: #source.get(#secret_name).map(|s| s.expose_secret().to_string())
```

Comment:
We should probably keep the Secrecy wrapper.

Reply:
Let's leave strings for now, I think we should design our own wrapper that's going to offer a bunch of other things like #11